changelog/12625.improvement.rst

```diff
@@ -0,0 +1,2 @@
+Conditionally ignore collection of setuptools artifacts dirnames only if the
+directories reside inside a setuptools project, i.e. `setup.cfg`, is present, etc.
```

Comment:
Please, enumerate all the files taken into consideration in this logic. Change notes are targeting the end-users who would not be able to guess. It might also be a good idea to mention the same somewhere in the regular docs, additionally.